### PR TITLE
Add is_unique for Index & MultiIndex

### PR DESCRIPTION
**Pull request by itholic**
This PR proposes `is_unique` for `Index` & `MultiIndex`.

Basically it's opposite to `has_duplicates`

```python
>>> idx = ks.Index([1, 5, 7, 7])
>>> idx.is_unique
False

>>> idx = ks.Index([1, 5, 7])
>>> idx.is_unique
True

>>> idx = ks.Index(["Watermelon", "Orange", "Apple",
...                 "Watermelon"])
>>> idx.is_unique
False

>>> idx = ks.Index(["Orange", "Apple",
...                 "Watermelon"])
>>> idx.is_unique
True
```

And fixed some tests to sync with pandas'.

### CODE REVIEW

File: databricks/koalas/indexes.py [hunk 1]
```diff
@@ -456,23 +456,56 @@ def has_duplicates(self) -> bool:
...
+        >>> idx.has_duplicates
+        False
         """
         sdf = self._internal.spark_frame.select(self.spark.column)
         scol = scol_for(sdf, sdf.columns[0])
 
         return sdf.select(F.count(scol) != F.countDistinct(scol)).first()[0]
 
+    @property
+    def is_unique(self):
```
Comment (HyukjinKwon) [inline databricks/koalas/indexes.py:483]: Can we add it to the docs?

Comment (HyukjinKwon) [inline databricks/koalas/indexes.py:483]: Let's also add the type hints `-> bool`

**Review by HyukjinKwon — APPROVED** [review]
Looks good otherwise.

File: databricks/koalas/indexes.py (hunk 1 above)
Comment (itholic) [reply databricks/koalas/indexes.py]: Oh, my mistake. Thanks !

Comment (itholic) [reply databricks/koalas/indexes.py]: Sounds good

**Pull request merged**